admin/modules/forum/management.php

```diff
@@ -1989,11 +1989,25 @@
 				$db->delete_query("forums", "CONCAT(',',parentlist,',') LIKE '%,$fid,%'");
 		}
 
-		$db->delete_query("threads", "fid='{$fid}' {$delquery}");
-		$db->delete_query("posts", "fid='{$fid}' {$delquery}");
+		require_once MYBB_ROOT.'inc/class_moderation.php';
+		$moderation = new Moderation();
+
+		$query = $db->simple_select("threads", "tid", "fid='{$fid}' {$delquery}");
+		while($tid = $db->fetch_field($query, 'tid'))
+		{
+			$moderation->delete_thread($tid);
+		}
+
+		$query = $db->simple_select('posts', 'pid', "fid='{$fid}' {$delquery}");
+		while($pid = $db->fetch_field($query, 'pid'))
+		{
+			$moderation->delete_post($pid);
+		}
```

Comment:
> That or replicate pretty much the moderation code for forum deletions.

We could replicate the code with some `IN` queries indeed, but I think pagination would be better so that plugins that hook into `delete_thread()` would do their stuff too.